stdlib/array.ml

```diff
@@ -292,3 +292,27 @@ let stable_sort cmp a =
 
 
 let fast_sort = stable_sort
+
+(** {6 Iterators} *)
+
+let to_iter a =
+  let next i =
+    if i < length a
```

Comment:
Surely, the check is needed. It would eliminate the repeated call to `length`, but at the expense of storing the length in the closure.  `length` is a built-in, and the array will be fetched from the closure in the common case.